### PR TITLE
Finish Implementing Select/MultiSelect #3166

### PR DESCRIPTION
**Pull request by GARY121github**
## refactor: improve generateEmptyFieldValue function

- Optimize handling of MultiSelect and Select cases
- Provide a default value for Select based on FieldSelectValue
- Enhance code readability and maintainability

### CODE REVIEW

**Comment by GARY121github** [comment]
hello @lucasbordeau ,
I am new to open source and I am still understanding the code base.
Can you please help me to understand the issue more clearly?

File: packages/twenty-front/src/modules/object-record/utils/generateEmptyFieldValue.ts [hunk 1]
```diff
@@ -49,8 +49,21 @@ export const generateEmptyFieldValue = (
       };
     }
 
-    case FieldMetadataType.MultiSelect:
-    case FieldMetadataType.Select: {
+   // Handling MultiSelect and Select in the default case
+    default: {
+      if (
+        fieldMetadataItem.type === FieldMetadataType.MultiSelect ||
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-record/utils/generateEmptyFieldValue.ts:55]: I am not sure to understand why you choose to handle this in the default section. This should stay in their own case

Comment (charlesBochet) [reply packages/twenty-front/src/modules/object-record/utils/generateEmptyFieldValue.ts:55]: Also multi select is not implemented yet on the backend so it should stay has non implemented

File: packages/twenty-front/src/modules/object-record/utils/generateEmptyFieldValue.ts [hunk 2]
```diff
@@ -49,8 +49,21 @@ export const generateEmptyFieldValue = (
...
+        fieldMetadataItem.type === FieldMetadataType.Select
+      ) {
+        const defaultSelectValue = {
+          color: 'DEFAULT_COLOR', // Replace with the actual default color
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-record/utils/generateEmptyFieldValue.ts:59]: let's take the first possible option color and first select option if the field is non nullable and leave it null if nullable

**Review by charlesBochet — CHANGES_REQUESTED** [review]
Thanks for the contribution, I have left comments!

**Comment by charlesBochet** [comment]
Also, could you please add a first jest test on this function?
And update the PR title :)

**Review by charlesBochet — COMMENTED** [review]
@GARY121github You have marked the comments as resolved but I don't see the related change? :)

**Comment by GARY121github** [comment]
I cannot understand the issue properly can you please guide me?
```
case FieldMetadataType.MultiSelect: {
    // Handle MultiSelect-specific logic
    if (!backendSupportsMultiSelect) {
      throw new Error('MultiSelect not implemented on the backend yet');
    }
    // Additional MultiSelect handling
    break;
  }
  case FieldMetadataType.Select: {
    // Handle Select-specific logic
    const defaultSelectValue = {
      color: 'DEFAULT_COLOR', // Replace with the actual default color
      label: '',
      __typename: 'FieldSelectValue',
    };
    return defaultSelectValue;
  }
  default: {
    throw new Error('Unhandled FieldMetadataType');
  }
```
Is this code completing your requirements?
I have separated them from the default case.

**Comment by charlesBochet** [comment]
@GARY121github It looks good but it's not what I see in your PR

**Comment by GARY121github** [comment]
> @GARY121github It looks good but it's not what I see in your PR

I have updated the code according to your explanation should I create

**Comment by charlesBochet** [comment]
Here is what I see:
image

**Comment by GARY121github** [comment]
@charlesBochet I have sent new PR can you please review and merge it

**Review by charlesBochet — APPROVED** [review]
Thank you @GARY121github 
Your code was referencing an undefined backendSupportsMultiSelect variable. I have removed it as the backend does not support it at all yet
Also, we have changed the typing of Select (it use to be an object label + color), it is now a scalar which simplifies it

**Pull request merged**